plugins/audio-s5gen2/fu-audio-s5gen2-device.c

```diff
@@ -256,16 +245,17 @@ fu_qc_s5gen2_device_cmd_validation(FuQcS5gen2Device *self, GError **error)
 	FuQcOpcode opcode;
 	guint16 delay_ms;
 	guint8 data[FU_STRUCT_QC_VALIDATION_SIZE] = {0};
+	gsize read_len;
 	g_autoptr(GByteArray) req = fu_struct_qc_validation_req_new();
 	g_autoptr(GByteArray) reply = NULL;
 
 	if (!fu_qc_s5gen2_device_msg_out(self, req->data, req->len, error))
 		return FALSE;
-	if (!fu_qc_s5gen2_device_msg_in(self, data, sizeof(data), error))
+	if (!fu_qc_s5gen2_device_msg_in(self, data, sizeof(data), &read_len, error))
 		return FALSE;
 
 	/* do not care about FU_QC_OPCODE_TRANSFER_COMPLETE_IND format */
-	reply = fu_struct_qc_validation_parse(data, sizeof(data), 0, error);
+	reply = fu_struct_qc_validation_parse(data, read_len, 0, error);
```

Comment:
ya, this looks wrong -- just let the parse use the entire buffer -- and check read_len before calling parse.

Reply:
but why?
All whose `_parse()` functions are using `fu_memchk_read()` to validate the boundaries which literally do the same, keeping in mind that buffer is set for particular request size.

Reply:
_parse() defines the second arg to be the length of the first parameter. If you do parse_bytes() instead then we only have the immutable blob of known size.

Lets go back a bit, _why_ are you passing in read_len here?

Reply:
`fu_qc_s5gen2_device_msg_in()` above returns the exact bytes read, less or equal `sizeof(data)`.
I hope everything goes well, but if the device returns lesser data for any reason (i.e. internal issue, for instance), I will catch it here, instead of parsing empty buffer possibly containing valid values (0 in u16be, for example).

Reply:
@hughsie I think it is better to leave as is. Allows catching issues with unexpected/incorrect message size -- proven during review rework.

Reply:
I can tell you now; _Coverity isn't going to like it_. `read_len` is "tainted" and can't be used to verify the buffer read without some kind of check first.

Reply:
ok. do you mind if I do additional check of `read_len` before to use it then? or you insist on `sizeof(data)` here, and below?

Reply:
btw what tool are you using for coverity scan?